packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -818,13 +818,44 @@ class CupertinoSliverNavigationBar extends StatefulWidget {
 // lose their own states.
 class _CupertinoSliverNavigationBarState extends State<CupertinoSliverNavigationBar> {
   late _NavigationBarStaticComponentsKeys keys;
+  ScrollableState? _scrollableState;
 
   @override
   void initState() {
     super.initState();
     keys = _NavigationBarStaticComponentsKeys();
   }
 
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    _scrollableState?.position.isScrollingNotifier.removeListener(_handleScrollChange);
```

Comment:
Using the `isScrollingNotifier` makes sense to me. It will only notify when scrolling starts and stops, which is exactly what is needed here. 👍  Using scroll notifications would mean a lot of notifications we would not need.